staging/src/k8s.io/apiserver/pkg/util/flowcontrol/metrics/timing_ratio_histogram.go

```diff
@@ -210,6 +210,20 @@ func (v *TimingRatioHistogramVec) NewForLabelValuesSafe(initialNumerator, initia
 	}
 }
 
+func (v *TimingRatioHistogramVec) NewForLabelValuesEfficient(initialNumerator, initialDenominator float64, labelValues []string) RatioedGauge {
+	tro, err := v.NewForLabelValuesChecked(initialNumerator, initialDenominator, labelValues)
+	if err == nil {
+		klog.V(3).InfoS("TimingRatioHistogramVec.NewForLabelValuesEfficient hit the efficient case", "fqName", v.FQName(), "labelValues", labelValues)
+		return tro
+	}
+	if !compbasemetrics.ErrIsNotRegistered(err) {
+		klog.ErrorS(err, "Failed to extract TimingRatioHistogramVec member, using noop instead", "vectorname", v.FQName(), "labelValues", labelValues)
+		return tro
+	}
+	klog.ErrorS(nil, "TimingRatioHistogramVec.NewForLabelValuesEfficient hit the inefficient case", "fqName", v.FQName(), "labelValues", labelValues)
+	panic(v.FQName())
```

Comment:
I'm not a fan of it.
If we want to prevent from hitting the inefficient path, what I would do is change the `getGauge` method to sth like:
```
io.once.Do(func() {
  if err := util.WaitUntil(..., func() (bool, error) {
    result, err := readWriteConcurrencyGaugeVec.NewForLabelValuesChecked(0, 1, io.labelValues)
    if err != nil {
      return false, nil  // retry
    }
    io.gauge = result
    return true, nil
  })
}
```

Reply:
This PR is for diagnosis, not merging.

Reply:
My expectation is a panic during server startup with a stack trace that shows the filters being constructed before metrics being registered.

Reply:
Does any of the CI jobs retain the kube-apiserver log?

Reply:
All of them do - just go to the artifacts from the prow job page.

Reply:
I tried https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/111424/pull-kubernetes-node-e2e-containerd/1551804854588936192/ and could not find a kube-apiserver log.

I looked at https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/111424/pull-kubernetes-e2e-gce-100-performance/1551804854643462144 and could not find a kube-apiserver.log